### DISCUSSION TITLE
Environment Variable Interpolating

### DISCUSSION THREAD

**Discussion by Vittitow** [nrwl/nx#8378] [General]
Is it possible?

We have OS level environment settings e.g. `DEV_SECRET`, `QA_SECRET`, `PROD_SECRET` etc. pre-set. 

We're using the run-command executor and passing in a .env file specific to the context of the app and environment e.g. `workspaceRoot/apps/my-app/.dev.env`. The CLI tool being spawned by the run-command executor is looking for the environment variable `SECRET` without the environment prefix. 

So, in this example we're attempting to map/interpolate `DEV_SECRET` into `SECRET`. We've attempted to do this via specifying `SECRET=$DEV_SECRET` in the `workspaceRoot/apps/my-app/.dev.env` file. However, `SECRET` is set to the literal string `$DEV_SECRET` rather than the actual value of that environment variable.

Is there a way around this other than customizing our targets themselves to do this?

**Comment by alfaproject** [comment]
I'm also looking for the ability to do this, but either I'm missing something or it's not possible. I was expecting that `{env:APP_STAGE}` would work, but alas it doesn't ):
